account_multicurrency_revaluation/report/currency_unrealized_report.py

```diff
@@ -22,10 +24,19 @@ def __init__(self, account):
         self.keys_to_sum = ['gl_foreign_balance', 'gl_currency_rate',
                             'gl_revaluated_balance', 'gl_balance',
                             'gl_ytd_balance']
+        # to check if 'currency' and the 'revaluation rate'
+        # is the same for all lines
+        self.keys_to_check = ['curr_name', 'gl_currency_rate']
```

Comment:
no need for such check, report is formed on journal and it's already on one currency

Reply:
Hello I thinks I should check id the currency is consistent cause if the account have no `currency_id` so the constraint is applied  [https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_move.py#L445](https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_move.py#L445)

@fclementic2c  am I wrong ?

Reply:
and constraint on which you point should breakup entry on creation for different currencies

Reply:
even if I have a account without Account currency no ?
![image](https://user-images.githubusercontent.com/32262135/53021912-81bf2700-345a-11e9-97c6-60eed51a0075.png)

Reply:
yes " if not line.currency_id " - it's  when during work they start using currency, but moves already exist without currency, 
at least I had a lot of pain when we tried to mixin it

Reply:
I'm not sure that the constraint is applied if we have no currency on the account here (`if line.account_id.currency_id:`), in this case no `ValidationError` is raised and we can mix currencies.

Reply:
I've check it, if we have no currency for account and trying create lines with currency we get error, so while you group lines under one account no need to worry about multiple currency